lib/config/default-config.js

```diff
@@ -46,7 +46,16 @@ exports.defaultConfig = [
             ".git/**"
         ],
         languageOptions: {
-            parser: "@/espree"
+            ecmaVersion: "latest",
+            sourceType: "module",
+            parser: "@/espree",
+            parserOptions: {}
+        }
```

Comment:
Updating the defaults as briefly discussed in #14588. I think it makes sense to assume people will be using ESM and always want the latest `ecmaVersion`.